overlord/snapstate/handlers.go

```diff
@@ -737,6 +738,31 @@ func (m *SnapManager) undoMountSnap(t *state.Task, _ *tomb.Tomb) error {
 	return m.backend.RemoveSnapDir(snapsup.placeInfo(), otherInstances)
 }
 
+// saveSnapDisabledServices queries for all currently disabled snap services and
+// adds those inside *snap.Info.
+func (m *SnapManager) saveSnapDisabledServices(snapst *SnapState, pb progress.Meter) error {
```

Comment:
The name doesn't reflect that we reach out to the backend/system to query the state.. (I'm terrible at naming things so take my suggestions with a grain of salt). Also, I feel a bit uneasy about snapst being modified internally, maybe a pattern like this could work, WDYT?

```
var err error
if snapst.LastActiveDisabledServices, err = m.querySnapDisabledSerices(snapst, pb); err != nil {
    return err
}
```

Reply:
yeah I felt weird about modifying snap state too, but we need to do this same thing in a number of places and I wanted to be extra careful about not committing erroneous results to state if we encountered an error so we can have safe retry. I thought it's okay to modify state since we are still in a method of the manager and not in the backend, but honestly I'm not sure. I would like to not use your suggestion and go with something a bit more safe like:
```go
var err error
var svcs []string
if svcs, err = m.querySnapDisabledSerices(snapst, pb); err != nil {
    return err
}
 
snapst.LastActiveDisabledServices = svcs
```

which ensures we don't inadvertently commit incorrect state

Reply:
the latter does look better    (also "save" would make me think this commits, which it cannot, so we would need a different name in the current form)

Reply:
okay I will adjust to do `querySnapDisabledServices` and also not have that function modify snapst, but have the caller do the modification to snapst